src/ferui-components/tree-view/internal-interfaces.ts

```diff
@@ -12,6 +12,7 @@ export interface TreeNode<T> {
   parent: TreeNode<T> | null;
```

Comment:
As we've discussed, the TreeNode<T> interface should be public because we can use it when we want to use the custom templates (the function return an instance of TreeNode<T> and we need to be able to access it.)

Reply:
Okay Jon, I moved the TreeNode interface out of the internal interfaces along with the TreeNodeEvent one cause like you said they'll be used by the developer.